src/core/messaging.cpp

```diff
@@ -28,13 +28,13 @@ void messaging_init() {
 }
 
 
-void send_message(const json &msg) {
+void send_message(const string type, const json &msg) {
```

Comment:
I thought this was more desirable than defining listeners for a generic `msg` type and relying on the json for a specific type, but I can do it that way if you prefer.